Add instructions for building packages

timoxley opened the pull request:
Currently there are no instructions on how to build the `react-vr` & `react-vr-web` packages. This adds a minimum level of documentation for someone with no knowledge to go from nothing to a working build. 

Ideally, this wouldn't be necessary and the install would happen automatically as part of the post-install script for the top-level package, but this is better than the current situation where there appears to be no documentation at all.

mikearmstrong001 commented:
Thanks so much of providing this and you are correct these steps should be documented.
I've provided some feedback on the minimum number of steps required and these will also work on window.

andrewimm commented:
Can we also be clearer that this is for building from source, and not for most developers? The lack of accompanying description might confuse some people inexperienced with the npm ecosystem.

iox commented:
In order to build from master, I had to install babel, rollup and some plugins. Maybe it would be worth to note it in these instructions too:

```
npm install -g babel-cli babel-preset-flow babel-preset-es2015 babel-preset-babili babel-plugin-external-helpers rollup-plugin-replace rollup-plugin-babel rollup
```

andrewimm commented:
@iox installing any global packages should not be necessary, as long as the dependencies have been installed in the appropriate folders by running the commands mentioned in this change.

I'll be publishing a bootstrap script today that simplifies this process as well, so that build-packages.js can detect whether the appropriate deps have been installed, and install them if they haven't yet.

timoxley commented:
> detect whether the appropriate deps have been installed, and install them if they haven't yet.

Note that both yarn and npm should do this automatically if you just try to install in the directory again.

andrewimm commented:
Just landed a script that bootstraps all the dependencies in ed7b291a1d7707b467792ef772044beecbf41e7b
You can run `node scripts/install-deps.js` on its own, or if you're building packages you can just run `node scripts/build-packages.js` directly. The latter will call into your package manager of choice and update any dependencies needed to build the packages.

timoxley commented:
Updated PR to simply point users at `node scripts/install-deps.js`.

> Can we also be clearer that this is for building from source, and not for most developers? 

Changed title of section from "Building" to "Building from Source" and added "Not recommended for most developers".

iox commented:
@andrewimm Thanks for the tip, I have given it another try today and I was able to build the packages with the new instructions.

Unfortunately, I seem to be missing something, as I am unable to run my project with the latest master (which I need to do to test the fix to #280).

* How does versioning work in this project? Master is version 1.3.0, but the "1.4.0-stable" branch is "6 commits ahead, 73 commits behind master."
* Any idea why my existing project will fail when using the packages built from master?:

![screenshot_20170828_092035](https://user-images.githubusercontent.com/452823/29763147-2f6dba3e-8bd2-11e7-9bd3-84d2f3a04f87.png)

andrewimm commented:
@iox sorry if there is any confusion. Master is always "unversioned," and usually has breaking changes ahead of the last version. The commits ahead / behind are due to the fact that minor releases are generated by cherry-picking bugfixes into release branches. Master itself contains a number of breaking changes which cannot be release into minor versions without violating semver. In more detail, our release process is to cut new releases directly from master until it contains changes that would require a major version bump. At that point, we follow the cherry-pick & release-branch strategy so that we can continue to release bugfixes without blocking on major version releases.

In this case, master is leading up to a 2.0.0 release, and has new dependencies on Three.js, React, and React Native. You can see these latest dependencies here: https://github.com/facebook/react-vr/blob/master/react-vr-cli/generators/package.json.generator.js#L27. To get the master branch to run, you'll need to update your project's package.json file to these versions.

iox commented:
Thank you very much @andrewimm . With your explanation, it makes much more sense. Now our project is running nicely with master!

I will prepare another pull request with the complete steps I'm following to work against master in order to test a new feature/fix.
